### ISSUE TITLE
FlowId filter inject tag in open tracing span

### ISSUE THREAD

**Issue by evangelion1204** [zalando/skipper#1516]
**Is your feature request related to a problem? Please describe.**
By tagging the spans it would enable people to do a search in their specific provider for a given flow id. Because Skipper is generating the id it would also be the perfect place to inject be. Especially helpful for people in 24/7 needing quickly map issues from skipper metrics to the actual tracing.

**Describe the solution you would like**
Enhancing the `flowid` filter to set the tag - an optional parameter for startup could be the name of the tag.

**Would you like to work on it?**
Yes, but no time

**Comment by lmineiro** [comment]
An alternative that could make sense as well would be to add another flowId `[Generator](https://github.com/zalando/skipper/blob/master/filters/flowid/filter.go#L18)` that uses as FlowID the TraceID. that FlowID generator would then reuse the current tracing identifier as the FlowID instead of the other way around. This is actually the long term strategy for Zalando.

**Comment by evangelion1204** [comment]
Sound reasonable - any plan on applying that change? Will that be a new filter or modify the existing one?

**Comment by szuecs** [comment]
@AlexanderYastrebov can we do this with templating maybe?
I am thinking of a header to [tracingTag](https://opensource.zalando.com/skipper/reference/filters/#tracingtag) case.
[FlowID](https://opensource.zalando.com/skipper/reference/filters/#flowid) is created or reused from HTTP requests and if we would make `getHeader`, such that the following would work, it would create a lot more value, than the original request would add:
```
 getHeader("X-Flow-Id", $flowid) -> tracingTag("my-flow-id", $flowid)
```

**Comment by AlexanderYastrebov** [comment]
@szuecs
If we enhance `tracingTag` to support template placeholders the solution would look like
```
* -> flowId("reuse") -> tracingTag("xflowid", "${request.header.X-Flow-Id}")
```

I have also noticed the following code
https://github.com/zalando/skipper/blob/5801bbb44c13d3018490f4cfd1d9e3a50d897913/proxy/proxy.go#L1454-L1456
but it seems there is a bug because `FlowIDTag` is defined as
https://github.com/zalando/skipper/blob/5801bbb44c13d3018490f4cfd1d9e3a50d897913/proxy/tracing.go#L11
I.e. it is supposed to work but overwrites(?) `component` tag if header is present.

The `setCommonSpanInfo` is called for both incoming request (`span.kind=server`) and backend request (`span.kind=client`). When header is absent, for incoming request it would not be able to set value generated by `flowId` filter because it is called before filter chain is invoked, for backend request it should work.

So I think we need to fix proxy flow id code or remove flowid support form `setCommonSpanInfo`.
Extending `tracingTag` to support template placeholders may be also useful for other tags besides flow id.

**Comment by AlexanderYastrebov** [comment]
Looks like a duplicate of https://github.com/zalando/skipper/issues/704

**Comment by AlexanderYastrebov** [comment]
Adding flow id tag in proxy code was done within #882 and broken by #1066 ([line](https://github.com/zalando/skipper/pull/1066/files#diff-cb531749aeef2d9124889e10dec27542ddebc8140a7862f406821297d9d74e7aR11))

**Comment by szuecs** [comment]
Thanks @AlexanderYastrebov

**Comment by AlexanderYastrebov** [comment]
Fixed by #1666
